src/optimum/intel/neural_compressor/utils.py

```diff
@@ -97,3 +102,61 @@ def _get_quantizable_ops_recursively(
         else:
             self._get_quantizable_ops_recursively(child, op_name, quantizable_ops)
 
+
+def remove_inputs_from_graph(gm_original: GraphModule, inputs_to_remove: List[str]) -> GraphModule:
+    """
+    Remove specified inputs from a GraphModule.
+
+    Args:
+        gm_original (:obj:`GraphModule`):
+            Original GraphModule.
+        inputs_to_remove (:obj:`List[str]`):
+            List of string specifying the name of the inputs to remove from the GraphModule.
+    Returns:
+        gm (:obj:`GraphModule`):
+            GraphModule with the removed inputs.
+    """
+    from transformers.utils.fx_transformations import deepcopy_graph
+
+    try:
+        gm = deepcopy_graph(gm_original)
+        gm.eval()
+    except Exception as e:
+        gm = gm_original
+        logger.warning(f"Deepcopy failed: {repr(e)}, model is modified inplace.")
+
+    graph = gm.graph
+    output_node = None
+    for node in graph.nodes:
+        if node.name == "output":
+            output_node = node
+            break
+
+    for node in graph.nodes:
+        if node.op == "placeholder" and node.target in inputs_to_remove:
+
+            def remove_users(node, output_node):
+                output_args = output_node.args[0] if output_node is not None else None
+
+                for user in list(node.users):
```

Comment:
Do you need to make node.users a list?

Reply:
This is done in order to make a copy of the keys of the dictionnary node.users as the latter is being modified